src/core/sensor/qgssensormodel.cpp

```diff
@@ -0,0 +1,267 @@
+                return tr( "Disconnected" );
+
+              case Qgis::DeviceConnectionStatus::Connecting:
+                return tr( "Connecting" );
+
+              case Qgis::DeviceConnectionStatus::Connected:
+                return sensor->data().lastValue.toString();
+            }
+          }
+          return QVariant();
+        }
+
+        default:
+          break;
+      }
+
+      return QVariant();
+    }
+
+    case SensorType:
+    {
+      QgsAbstractSensor *sensor = mSensorManager->sensor( mSensorIds[index.row()] );
+      return sensor ? sensor->type() : QString();
+    }
+
+    case SensorId:
+    {
+      return mSensorIds[index.row()];
+    }
+
+    case SensorName:
+    {
+      QgsAbstractSensor *sensor = mSensorManager->sensor( mSensorIds[index.row()] );
+      return sensor ? sensor->name() : QString();
+    }
+
+    case SensorStatus:
+    {
+      QgsAbstractSensor *sensor = mSensorManager->sensor( mSensorIds[index.row()] );
+      return QVariant::fromValue<Qgis::DeviceConnectionStatus>( sensor ? sensor->status() : Qgis::DeviceConnectionStatus::Disconnected );
+    }
+
+    case SensorLastValue:
+    {
+      QgsAbstractSensor *sensor = mSensorManager->sensor( mSensorIds[index.row()] );
+      return sensor ? sensor->data().lastValue : QVariant();
+    }
+
+    case SensorLastTimestamp:
+    {
+      QgsAbstractSensor *sensor = mSensorManager->sensor( mSensorIds[index.row()] );
+      return sensor ? sensor->data().lastTimestamp : QDateTime();
+    }
+
+    case Sensor:
+    {
+      return QVariant::fromValue<QgsAbstractSensor *>( mSensorManager->sensor( mSensorIds[index.row()] ) );
+    }
+
+    default:
+      return QVariant();
+  }
+#ifndef _MSC_VER // avoid warning
+  return QVariant();  // avoid warning
+#endif
+}
+
+bool QgsSensorModel::setData( const QModelIndex &index, const QVariant &value, int role )
+{
+  if ( index.row() < 0 || index.row() >= rowCount( QModelIndex() ) || role != Qt::EditRole )
+    return false;
+
+  switch ( index.column() )
+  {
+    case Name:
+    {
+      QgsAbstractSensor *sensor = mSensorManager->sensor( mSensorIds[index.row()] );
+      if ( sensor )
+      {
+        sensor->setName( value.toString() );
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+Qt::ItemFlags QgsSensorModel::flags( const QModelIndex &index ) const
+{
+  Qt::ItemFlags flags = QAbstractItemModel::flags( index );
+  if ( index.isValid() && index.column() == Name )
+  {
+    return flags | Qt::ItemIsEditable;
+  }
+  else
+  {
+    return flags;
+  }
+}
+
+QVariant QgsSensorModel::headerData( int section, Qt::Orientation orientation, int role ) const
+{
+  if ( role == Qt::DisplayRole )
+  {
+    if ( orientation == Qt::Vertical ) //row
+    {
+      return QVariant( section );
+    }
+    else
+    {
+      switch ( section )
+      {
+        case Name:
+          return QVariant( tr( "Name" ) );
+
+        case LastValue:
+          return QVariant( tr( "Last value" ) );
```

Comment:
Will need double-checking, but pretty sure our convention is for Title Case for these headers
```suggestion
          return QVariant( tr( "Last Value" ) );
```